docs/sdk/src/guides/enable_pov_reclaim.rs

```diff
@@ -0,0 +1,81 @@
+//! during [benchmarking](`crate::reference_docs::frame_benchmarking_weight`) represents the worst
+//! case. In reality, runtime operations often consume less space in the storage proof. PoV reclaim
+//! offers a mechanism to reclaim the difference between the benchmarked worst-case and the real
+//! proof-size consumption.
+//!
+//!
+//! # How to enable PoV reclaim
+//! ## 1. Add the host function to your node
+//!
+//! To reclaim excess storage weight, a parachain runtime needs the
+//! ability to fetch the size of the storage proof from the node. The reclaim
+//! mechanism uses the
+//! [storage_proof_size](`cumulus_primitives_proof_size_hostfunction::storage_proof_size`)
+//! hostfunction for this purpose. For convenience, cumulus provides
+//! [ParachainHostFunctions](`cumulus_client_service::ParachainHostFunctions`), a set of
+//! hostfunctions typically used by cumulus-based parachains. In the binary crate of your parachain,
+//! find the instantiation of the [WasmExecutor](`sc_executor::WasmExecutor`) and set the correct
+//! generic type.
+//!
+//! This example from the parachain-template shows a type definition that includes the correct
+//! hostfunctions.
+#![doc = docify::embed!("../../templates/parachain/node/src/service.rs", wasm_executor)]
+//!
+//! > **Note:**
+//! >
+//! > If you see error `runtime requires function imports which are not present on the host:
+//! > 'env:ext_storage_proof_size_storage_proof_size_version_1'`, it is likely
+//! > that this step in the guide was not set up correctly.
+//!
+//! ## 2. Enable storage proof recording during import
+//!
+//! The reclaim mechanism reads the size of the currently recorded storage proof multiple times
+//! during block authoring and block import. Proof recording during authoring is already enabled on
+//! parachains. You must also ensure that storage proof recording is enabled during block import.
+//! Find where your node builds the fundamental substrate components by calling
+//! [new_full_parts](`sc_service::new_full_parts`). Replace this
+//! with [new_full_parts_record_import](`sc_service::new_full_parts_record_import`) and
+//! pass `true` as the last parameter to enable import recording.
+#![doc = docify::embed!("../../templates/parachain/node/src/service.rs", component_instantiation)]
+//!
+//! > **Note:**
+//! >
+//! > If you see error `Storage root must match that calculated.` during block import, it is likely
+//! > that this step in the guide was not
+//! > set up correctly.
+//!
+//! ## 3. Add the SignedExtension to your runtime
+//!
+//! In your runtime, you will find a list of SignedExtensions.
+//! To enable the reclaiming,
+//! add [StorageWeightReclaim](`cumulus_primitives_storage_weight_reclaim::StorageWeightReclaim`)
+//! to that list. The extension will check the size of the storage proof before and after an
+//! extrinsic execution. It reclaims the difference between the calculated size and the benchmarked
+//! size.
```

Comment:
One note please if it matters where to add it in the list of SEs. I assume at the end, but could be mentioned.

Reply:
I was thinking about it during implementation and came to the conclusion that the order should not be important. But maybe I failed. Why do you think it should be at the end?

Reply:
>To answer my own question, when it is placed at the end it will be able to operate more accurately because storage changes performed by the other SignedExtensions can be reflected as well.

Yes that is what i thought. Having it before that could miss some of the following consumed weight. So i guess end is best?  

I would still like to use it to actually refund the overspent fee payment. Since we will need to bump the PoV weights a bit, and then refunding that amount would be nice.

Reply:
> I would still like to use it to actually refund the overspent fee payment. Since we will need to bump the PoV weights a bit, and then refunding that amount would be nice. I can try to do it.

Slightly offtopic for this PR, but do you have an idea how we could deal with the MEV implications? Imagine multiple tx load some expensive storage item. Only the first executed would be charged since the other ones are not growing the proof size at all. Collators could benefit from reordering the transactions, which we wanted to avoid.

Reply:
> Only the first executed would be charged since the other ones are not growing the proof size at all. Collators could benefit from reordering the transactions, which we wanted to avoid.

Yea... i mentioned MEV and secondary market at my last Sub0 talk as consequence of this weight fee reclaim.  
I think it needs a wider discussion. There are probably some tricks that can be done, did not check how other ecosystems solve this. cc @kianenigma